api/src/main/java/graphql/nadel/hooks/ServiceExecutionHooks.java

```diff
@@ -53,11 +54,10 @@ default NewVariableValue visitArgumentValueInQuery(HooksVisitArgumentValueEnviro
      * Called to resolve the service that should be used to fetch data for a field that uses dynamic service resolution.
      *
      * @param services a list of all services registered on Nadel
-     * @param fieldName the name of the field
-     * @param arguments GraphQL arguments that were passed when querying the field
+     * @param executionStepInfo
      * @return the Service that should be used to fetch data for that field or an error that was raised when trying to resolve the service.
      */
-    default ServiceOrError resolveServiceForField(List<Service> services, String fieldName, Map<String, Object> arguments) {
+    default ServiceOrError resolveServiceForField(List<Service> services, ExecutionStepInfo executionStepInfo) {
```

Comment:
Getting a `ExecutionStepInfo` which contains everything the hook method will need to create a `GraphQLError`.